### DISCUSSION TITLE
Mockoon crashes when operating with larger json files

### DISCUSSION THREAD

**Discussion by chirantha89** [mockoon/mockoon#448] [Help] [answered]
Hi,

I'm using Mockoon 1.11.0 version. I have an exported json file of Mockoon 1.8.0 version. The json file is around 66MB of size.
When I import this file to 1.11.0 version, it's importing without giving any errors. But when I tried to mock new end points for my imported project, mockoon got crashed.
What could be the reason for this?
Your help is really appreciate on this.

Regards

**Comment by 255kb** [comment]
Hi, while Mockoon is quite performant, it has not been designed to be used with big data files. It may have silently failed during import. Performances are definitely something we would like to work on this year. 
If you want, I can have a look and test directly with your data file, if you are able to share it (support@mockoon.com).
Thanks

**Reply by chirantha89** [reply]
Hi,
Thank you very much for your support. I'm really sorry that I won't be able to share the data file due to GDPR regulations.
Is there any workaround for this issue? Can't we increase the heap memory of Mockoon?

Regards,

**Reply by 255kb** [reply]
Before tuning Electron memory options I would like to investigate the cause of the slowness/crash. Could you please describe a bit the structure of your data file? Number of environments, routes, using body editor or external files, etc. 
Also, are you importing data in Mockoon's format or OpenAPI/Swagger?

**Reply by chirantha89** [reply]
Hi,
I'm importing data in Mockoon's format. There is only one environment as of now.  And there are 3 routes.
Each and every route is having around 25 mock responses and most of the mock responses are having more than 100,000 lines.

Regards,

**Comment by 255kb** [comment, marked as answer]
@chirantha89 We will track this problem in the following issue and try to reproduce it: https://github.com/mockoon/mockoon/issues/450
Let us know if you can provide more information about the error you got, if any. Also if you can find any trace of this in your logs it would help a lot (open app data folder with menu Tools->Show app data folder, and navigate to `./mockoon/logs/main.log` or `./mockoon/logs/renderer.log`)

**Reply by chirantha89** [reply]
Hi,
I really appreciate that you have started looking on this issues. It didn't throw any error. I checked in the log file as well. There is no trace. Here I have attached a screenshot of Mockoon when it got crashed.

![Screenshot_1](https://user-images.githubusercontent.com/59163400/114994307-4691d480-9eba-11eb-856e-358136eb8bb3.png)

**Reply by 255kb** [reply]
Indeed that's quite a serious crash. Did this occurred during the export or the import?

**Reply by chirantha89** [reply]
Hi,

No this happened while doing operations for the imported json file like adding more mock responses, adding rules, etc.

Regards,

**Reply by 255kb** [reply]
ok, thank you for the details.

**Reply by chirantha89** [reply]
Hi,

Any update on this?

Regards

**Reply by 255kb** [reply]
Hi @chirantha89 
I couldn't reproduce this until now. I created a setup similar to yours, with one environment, up to 10 routes, 25 responses each with bodies containing JSON of 100 000 lines (~320MB data file). The UI became a bit laggy but I could still work with it and export/reimport. I played a bit more with the interface and import/export features but I couldn't reproduce the crash you encountered.

Here is my feeling on this: 
- without being able to reproduce this bug it's very hard to pinpoint the cause 
- I am not sure that increasing the memory limit would solve the problem as I see the app processes using up to 8GB of RAM during heavy tasks (import/export) (which indicate that the limit is already quite high).
- Having huge JSON bodies in the editor is not the most common use case, and you may want to use the following alternatives: 
  - store the data in separated JSON files and serve them instead of using the body editor, 
  - use the templating system to generate the data, so only a small template is stored, something like this:
    ```
    [
      {{#repeat 5000}}
      {
        "firstname": "{faker 'name.firstName'}}",
        "lastname": "{{faker 'name.lastName'}}"
      }
      {{/repeat}}
    ]
    ```
For these reasons, I am going to close the issue as non reproducible, at least for now. I will still keep an eye on it, as I also plan to improve the performances by using service workers for some tasks.
